File: AFSwinx/src/main/java/com/tomscz/afswinx/rest/connection/XMLParser.java
package com.tomscz.afswinx.rest.connection;

import com.tomscz.afswinx.common.ParameterMissingException;
import org.w3c.dom.Document;

/**
 * This interface specify behavior of XMLParses.
 * 
 * @author Martin Tomasek (martin@toms-cz.com)
 * 
 * @since 1.0.0.
 */
public interface XMLParser {

    /**
     * This method parse XML document.
     * 
     * @param documentToParse document which should be parsed
     * @return data of document. Typically as object of generic type.
     * @throws ParameterMissingException if some parameters during expression evaluation is missing
     */
    public <T> T parseDocument(Document documentToParse) throws ParameterMissingException;

}

File: AFMobile/AFWindowsPhone/AFWindowsPhone/parsers/XMLParser.cs
using AFWindowsPhone.rest.connection;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;
using Windows.Data.Xml.Dom;

namespace AFWindowsPhone.parsers
{
    interface XMLParser
    {
     
        AFSwinxConnectionPack parseDocument(XmlDocument documentToParse);
    }
}
